src/java_bytecode/Makefile

```diff
@@ -36,10 +36,52 @@ INCLUDES= -I ..
 include ../config.inc
 include ../common
 
-CLEANFILES = java_bytecode$(LIBEXT)
+
+SOURCE_DIR := library/src
+BINARY_DIR := library/classes
+
+FIND := find
+
+JAVAC := javac
+JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR) -XDignore.symbol.file
+
+CLASSPATH := SOURCE_DIR
+
+ALL_JAVAS := $(wildcard $(SOURCE_DIR)/*/*.java $(SOURCE_DIR)/*/*/*.java $(SOURCE_DIR)/*/*/*/*.java) #$(shell find $(SOURCE_DIR) -name '*.java')
+ALL_CLASSES := $(patsubst $(SOURCE_DIR)/%.java,$(BINARY_DIR)/%.class,$(ALL_JAVAS))
+
+$(BINARY_DIR):
+	mkdir -p $(BINARY_DIR)
+
+.SUFFIXES: .java .class
+
+$(BINARY_DIR)/%.class : $(SOURCE_DIR)/%.java $(BINARY_DIR)
+	$(JAVAC) $(JFLAGS) $(patsubst $(BINARY_DIR)/%.class,$(SOURCE_DIR)/%.java,$@)
+
+JAR := jar
+JARFLAGS := -cf
+
+.NOTPARALLEL: core-models.jar
```

Comment:
The use of `.NOTPARALLEL` makes the execution of make on the **entire** folder to be not parallel, which is an unacceptable performance hit. @dcattaruzza will create a new `Makefile` inside of the `library` folder and make that one sequential.

Reply:
done